kernel/posix/semaphore.c

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <pthread.h>
+#include <errno.h>
+
+/**
+ * @brief destroy semaphore.
+ *
+ * FIXME: EBUSY is not taken care as of now
+ *
+ * see IEEE 1003.1
+ */
+int sem_destroy(sem_t *semaphore)
+{
+	if (semaphore == NULL) {
+		errno = EINVAL;
+		return -1;
+	}
+
+	k_sem_reset(semaphore);
+
+	return 0;
+}
+
+/**
+ * @brief Get value of semaphore
+ *
+ * See IEEE 1003.1
+ */
+int sem_getvalue(sem_t *semaphore, int *value)
+{
+	*value = (int) k_sem_count_get(semaphore);
+
+	return 0;
+}
+/**
+ * @brief initialize semaphore
+ *
+ * While using this API, pass pshare value as 0 as feature
```

Comment:
which feature is not supported ?  something like - API only accepts 0 to share semphore among threads. Values greater than 0 are invalid as zephyr does not support processes.